src/target/riscv/riscv-013.c

```diff
@@ -686,6 +696,27 @@ static int dmi_write_exec(struct target *target, uint32_t address,
 	return dmi_op(target, NULL, NULL, DMI_OP_WRITE, address, value, true, ensure_success);
 }
 
+static bool check_dbgbase_exists(struct target *target)
+{
+	uint32_t dbgbase_to_find = target->dbgbase;
+	/* Start with the first debug module */
+	uint32_t nextdm = 0;
+
+	LOG_TARGET_DEBUG(target, "Searching for DM with DMI base address (dbgbase) = 0x%x", target->dbgbase);
+	while (1) {
```

Comment:
Can you limit this loop to 100 DMs or so? We should try to avoid an infinite loop if the hardware is broken.

Ideally we'd check for duplicates, but that seems more complicated to implement and unnecessary.

Reply:
done